simulator/modules/core/t/unit_test.cpp

```diff
@@ -171,3 +171,23 @@ TEST_CASE( "Torture_Test: Perf_Sim, RISC-V 32 simple trace")
     CHECK( sim->run_no_limit() == Trap::HALT);
     CHECK( sim->get_exit_code() == 0);
 }
+
+TEST_CASE( "Perf_sim: Syscall flushes pipeline")
+{
+    std::istringstream input( "4\n8\n");
+    std::ostringstream output;
+    auto sim = CycleAccurateSimulator::create_simulator( "riscv32");
+    auto mem = FuncMemory::create_default_hierarchied_memory();
+    sim->set_memory( mem);
+    auto kernel = create_mars_kernel( input, output, std::cerr);
+    kernel->set_simulator( sim);
+    kernel->connect_memory( mem);
+    kernel->connect_exception_handler();
+    kernel->load_file( TEST_PATH "/rv32-scall");
```

Comment:
That's too complicated program for this test case.
Consider your follower next year making other feature and suddenly this unit test starts to fail. He or she will assume it is related to system calls and pipeline flushes, and spent effort investigating it, while the real issue happens in another part of the execution (e.g. addition). I know it sounds hypothetical, but it's better to prevent problems than solve it, especially in our project when the ownership is lost with a new academical year.

The bottom line is: we cannot create a unit test for the feature, but we should at least approximate to it.